src/Symfony/Component/Form/Tests/Extension/Core/Type/UrlTypeTest.php

```diff
@@ -95,4 +103,47 @@ public function testSubmitNullUsesDefaultEmptyData($emptyData = 'empty', $expect
         $this->assertSame($expectedData, $form->getNormData());
         $this->assertSame($expectedData, $form->getData());
     }
+
+    /**
+     * @todo remove in 7.0
+     */
+    public function testSubmitNullReturnsNullWithEmptyDataAsString()
+    {
+        $form = $this->factory->create(static::TESTED_TYPE, 'name', [
+            'default_protocol' => 'http',
+            'empty_data' => '',
+        ]);
+
+        $form->submit(null);
+        $this->assertSame('', $form->getData());
+        $this->assertSame('', $form->getNormData());
+        $this->assertSame('', $form->getViewData());
+    }
+
+    /**
+     * @dataProvider provideZeros
+     *
+     * @todo remove in 7.0
```

Comment:
same here

Reply:
Fixed in https://github.com/symfony/symfony/pull/50922/commits/7956552402c76d490f21f1a0a15159f82b26ea9c